src/Sylius/Behat/Service/SessionService.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Behat\Service;
+
+use Behat\Mink\Mink;
+
+final class SessionService implements SessionServiceInterface
+{
+    private Mink $mink;
+
+    private SharedStorageInterface $sharedStorage;
+
+    private SecurityServiceInterface $securityService;
+
+    private const SESSION_CHROME_HEADLESS_SECOND = 'chrome_headless_second_session';
+
+    public function __construct(Mink $mink, SharedStorageInterface $sharedStorage, SecurityServiceInterface $securityService)
+    {
+        $this->mink = $mink;
+        $this->sharedStorage = $sharedStorage;
+        $this->securityService = $securityService;
+    }
+
+    public function changeSession(): void
+    {
+        $sessionName = self::SESSION_CHROME_HEADLESS_SECOND;
+
+        $this->savePreviousSession();
+
+        $this->mink->setDefaultSessionName($sessionName);
+
+        $this->mink->restartSessions();
```

Comment:
I believe we can extract this as well to some `saveAndRestartSession` function (as it's duplicated here and in the next method)